### PR TITLE
Added automation attributes for profile settings, change password and email modals

### PR DESCRIPTION
**Pull request by 0vid1u**

### CODE REVIEW

File: src/components/Modal/index.jsx [hunk 1]
```diff
@@ -29,13 +29,13 @@ const Modal = ({ children, isOpen, toggleModal, modalTitleId }) => {
       shouldCloseOnOverlayClick={true}
     >
       <div>
-        <CloseButton onClick={() => {
+        <CloseButton id="close-modal-btn" onClick={() => {
```
Comment (MihailGorceagEndava) [inline src/components/Modal/index.jsx:32]: here should be **data-test** attribute but not **id**

File: src/components/Modal/index.jsx [hunk 2]
```diff
@@ -29,13 +29,13 @@ const Modal = ({ children, isOpen, toggleModal, modalTitleId }) => {
...
           return toggleModal(false)
         }}><CloseIcon></CloseIcon></CloseButton>
       </div>
       <div style={{ width: '416px' }}>
         <div>
-          <ModalLabelContainer >
+          <ModalLabelContainer id="modal-title">
```
Comment (MihailGorceagEndava) [inline src/components/Modal/index.jsx:38]: **data-test** but not **id**

File: src/pages/ProfileManagement/components/ChangeEmailModalForm.js [hunk 3]
```diff
@@ -60,23 +60,25 @@ const ChangeEmailModalForm = ({ toggleModal }) => {
...
     </Modal>}
-    <Input
-      field="email"
-      value={email}
-      label="email"
-      onChange={handleEmailChange}
-      placeholder="emailPlaceholder"
-      error={error}
-    />
+    <div data-test="form-field">
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangeEmailModalForm.js:63]: remove div, and provide dataTest prop to Input component

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 4]
```diff
@@ -85,39 +85,45 @@ const ChangePasswordModalForm = ({toggleModal}) => {
...
-    />
-    <PasswordInput
-      field="confirmNewPassword"
-      value={confirmNewPassword}
-      label="confirmNewPassword"
-      onChange={handleConfirmNewPasswordChange}
-      placeholder="confirmNewPassword"
-      error={error}
-    />
+    <div dataTest="form-field">
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:88]: remove div and provide data-test prop to PasswordInput

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 5]
```diff
@@ -85,39 +85,45 @@ const ChangePasswordModalForm = ({toggleModal}) => {
...
+      <PasswordInput
+        field="password"
+        value={currentPassword}
+        label="password"
+        onChange={handleOldPasswordChange}
+        placeholder="enterCurrentPassword"
+        error={error}
+      />
+    </div>
+    <div dataTest="form-field">
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:98]: remove div, and provide data-test attribute to PasswordInput

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 6]
```diff
@@ -85,39 +85,45 @@ const ChangePasswordModalForm = ({toggleModal}) => {
...
+      <PasswordInput
+        field="newPassword"
+        value={newPassword}
+        label="newPassword"
+        onChange={handleNewPasswordChange}
+        placeholder="enterNewPassword"
+        error={error}
+      />
+    </div>
+    <div dataTest="form-field">
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:108]: remove div, and provide data-test attribute to PasswordInput

File: src/pages/ProfileManagement/components/ProfileDataForm.js [hunk 7]
```diff
@@ -65,15 +65,15 @@ const ProfileDataForm = (profile) => {
...
+        <div data-test="profile-icon" style={{marginBottom: '64px'}}>
           <ProfileIcon
             user={profileData}
           />
         </div>
       </Col>
     </Row>
     <Row>
-      <Col md={6}>
+      <Col dataTest="form-field" md={6}>
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ProfileDataForm.js:76]: should be data-test here, but not dataTest

File: src/pages/ProfileManagement/components/ProfileDataForm.js [hunk 8]
```diff
@@ -83,7 +83,7 @@ const ProfileDataForm = (profile) => {
           placeholder="firstName"
         />
       </Col>
-      <Col md={6}>
+      <Col dataTest="form-field" md={6}>
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ProfileDataForm.js:86]: should be data-test here, but not dataTest

File: src/pages/ProfileManagement/components/ProfileDataForm.js [hunk 9]
```diff
@@ -94,7 +94,7 @@ const ProfileDataForm = (profile) => {
       </Col>
     </Row>
     <Row>
-      <Col md={6}>
+      <Col dataTest="form-field" md={6}>
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ProfileDataForm.js:97]: should be data-test here, but not dataTest

File: src/pages/ProfileManagement/components/ProfileDataForm.js [hunk 10]
```diff
@@ -107,7 +107,7 @@ const ProfileDataForm = (profile) => {
           menuPlacement="bottom"
         />
       </Col>
-      <Col md={6}>
+      <Col dataTest="form-field" md={6}>
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ProfileDataForm.js:110]: should be data-test here, but not dataTest

File: src/components/Modal/index.jsx (hunk 1 above)
Comment (0vid1u) [reply src/components/Modal/index.jsx:32]: done

File: src/components/Modal/index.jsx (hunk 2 above)
Comment (0vid1u) [reply src/components/Modal/index.jsx:38]: done

File: src/pages/ProfileManagement/components/ChangeEmailModalForm.js (hunk 3 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangeEmailModalForm.js:63]: done

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js (hunk 4 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangePasswordModalForm.js:88]: done

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js (hunk 5 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangePasswordModalForm.js:98]: done

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js (hunk 6 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangePasswordModalForm.js:108]: done

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 11]
```diff
@@ -86,6 +86,7 @@ const ChangePasswordModalForm = ({toggleModal}) => {
 
   return (<>
     <PasswordInput
+      dataTest="form-field"
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:89]: I think here you would like to have data-test='currentPassword'

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 12]
```diff
@@ -94,6 +95,7 @@ const ChangePasswordModalForm = ({toggleModal}) => {
       error={error}
     />
     <PasswordInput
+      dataTest="form-field"
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:98]: I think here you would like to have data-test='newPassword'

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js [hunk 13]
```diff
@@ -102,6 +104,7 @@ const ChangePasswordModalForm = ({toggleModal}) => {
...
```
Comment (MihailGorceagEndava) [inline src/pages/ProfileManagement/components/ChangePasswordModalForm.js:107]: I think here you would like to have data-test='confirmNewPassword'

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js (hunk 11 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangePasswordModalForm.js:89]: that's a valid case, but we have implemented a generic approach to interact with web forms and we need to have the same selector for all form fields

File: src/pages/ProfileManagement/components/ChangePasswordModalForm.js (hunk 12 above)
Comment (0vid1u) [reply src/pages/ProfileManagement/components/ChangePasswordModalForm.js:98]: It's enough to mark it as form field

**Pull request closed without merging**